include/swift/AST/Decl.h

```diff
@@ -1161,12 +1168,34 @@ class ImportDecl final : public Decl,
     return static_cast<ImportKind>(Bits.ImportDecl.ImportKind);
   }
 
-  ImportPath getImportPath() const {
-    return ImportPath({ getTrailingObjects<ImportPath::Element>(),
-                        static_cast<size_t>(Bits.ImportDecl.NumPathElements) });
+  /// Retrieves the full import path.
+  /// \param outRealModuleName An ImportPath builder to write the real module name to if module aliasing
+  ///        was used. For example, if '-module-alias Foo=Bar' was passed, the real name 'Bar' will be written
+  ///        to it, which corresponds to 'import Foo' in source file.
+  /// \returns An ImportPath corresponding to this import decl. If module aliasing was used, 'Foo' will be in the
+  ///          returned path instead of 'Bar' using the above example.
+  ImportPath getImportPath(ImportPath::Builder *outRealModuleName = nullptr) const {
+    auto path = ImportPath({ getTrailingObjects<ImportPath::Element>(),
+      static_cast<size_t>(Bits.ImportDecl.NumPathElements) });;
+
+    if (outRealModuleName != nullptr && !RealModuleName.empty()) {
+      for (auto elem: path) {
+        if (outRealModuleName->empty()) {
+          // Add the real module name instead of its alias
+          outRealModuleName->push_back(RealModuleName);
+        } else {
+          // Add the rest if any (access path elements)
+          outRealModuleName->push_back(elem.Item);
+        }
+      }
+    }
+    return path;
   }
 
-  ImportPath::Module getModulePath() const {
+  ImportPath::Module getModulePath(ImportPath::Module::Builder *outRealModuleName = nullptr) const {
+    if (outRealModuleName != nullptr && !RealModuleName.empty()) {
+      outRealModuleName->push_back(RealModuleName);
+    }
```

Comment:
If you take my suggestion above, this would probably become:

```suggestion
  /// Retrieves the part of the import path that contains the module name,
  /// as written in the source code.
  /// 
  /// \returns A \c ModulePath corresponding to this import decl. If module
  ///          aliasing was used, this will contain the aliased name of the module;
  ///          for instance, if you wrote 'import Foo' but passed
  ///          '-module-alias Foo=Bar', this module path will include 'Foo'. This
  ///          return value is always owned by the AST context, so it can be
  ///          persisted.
  ImportPath::Module getModulePath() const {
    return getImportPath().getModulePath(getImportKind());
  }

  /// Retrieves the part of the import path that contains the module name,
  /// replacing any module aliases with real names.
  /// 
  /// \param scratch An \c ImportPath::Builder which may, if necessary, be used to
  ///        construct the return value. It may go unused, so you should not try to
  ///        read the result from it; use the return value instead.
  /// \returns An \c ImportPath corresponding to this import decl. If module
  ///          aliasing was used, this will contain the real name of the module;
  ///          for instance, if you wrote 'import Foo' but passed
  ///          '-module-alias Foo=Bar', this import path will include 'Bar'. This
  ///          return value may be owned by \p scratch, so it should not be used
  ///          after \p scratch is destroyed.
  ImportPath::Module getRealModulePath(ImportPath::Builder &scratch) const {
    return getRealImportPath(scratch).getModulePath(getImportKind());
```